release-21.2: changefeedccl: mark kvcoord sendError as retryable

miretskiy opened the pull request:
Backport 1/1 commits from #75517.

/cc @cockroachdb/release

---

During a cluster upgrade it may be that no replicas can be sent to and
we receive a "failed to send RPC" error.  This patch allows the
changefeed to retry in that instance.

Fixes: https://github.com/cockroachdb/cockroach/issues/72616

Release note (bug fix): changefeeds retry instead of fail on RPC send failure

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/76988)
<!-- Reviewable:end -->

Commit:
changefeedccl: mark kvfeed sendError as retryable

During a cluster upgrade it may be that no replicas can be sent to and
we receive a "failed to send RPC" error.  This patch allows the
changefeed to retry in that instance.

Release note (bug fix): changefeeds retry instead of fail on RPC send failure
